File: tarl/adapters/go/tarl.go
package tarl

type TARL struct {
    Version     string   `json:"version"`
    Intent      string   `json:"intent"`
    Scope       string   `json:"scope"`
    Authority   string   `json:"authority"`
    Constraints []string `json:"constraints"`
}

func New(intent, scope, authority string, constraints []string) TARL {
    return TARL{
        Version: "2.0",
        Intent: intent,
        Scope: scope,
        Authority: authority,
        Constraints: constraints,
    }
}

File: tarl/adapters/java/TARL.java
public final class TARL {
    public final String version = "2.0";
    public final String intent;
    public final String scope;
    public final String authority;
    public final String[] constraints;

    public TARL(String intent, String scope, String authority, String[] constraints) {
        this.intent = intent;
        this.scope = scope;
        this.authority = authority;
        this.constraints = constraints;
    }
}
